src/legacy/core_plugins/vis_type_timeseries/public/visualizations/views/timeseries/utils/theme.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import colorJS from 'color';
+import { Theme, LIGHT_THEME, DARK_THEME } from '@elastic/charts';
+
+function computeRelativeLuminosity(rgb: string) {
+  return colorJS(rgb).luminosity();
+}
+
+function getBrightestRelativeLum(rgb1: string, rgb2: string) {
+  const relLum1 = computeRelativeLuminosity(rgb1);
+  const relLum2 = computeRelativeLuminosity(rgb2);
+  return Math.max(relLum1, relLum2);
+}
+
+function getDarkestRelativeLum(rgb1: string, rgb2: string) {
+  const relLum1 = computeRelativeLuminosity(rgb1);
+  const relLum2 = computeRelativeLuminosity(rgb2);
+  return Math.min(relLum1, relLum2);
+}
+
+function computeContrast(rgb1: string, rgb2: string) {
+  const brightest = getBrightestRelativeLum(rgb1, rgb2);
+  const darkest = getDarkestRelativeLum(rgb1, rgb2);
+  return (brightest + 0.05) / (darkest + 0.05);
+}
+
+function getAAARelativeLum(bgColor: string, fgColor: string, ratio = 7) {
+  const relLum1 = computeRelativeLuminosity(bgColor);
+  const relLum2 = computeRelativeLuminosity(fgColor);
+  if (relLum1 > relLum2) {
+    // relLum1 is brighter, relLum2 is darker
+    return (relLum1 + 0.05 - ratio * 0.05) / ratio;
+  } else {
+    // relLum1 is darker, relLum2 is brighter
+    return Math.min(ratio * (relLum1 + 0.05) - 0.05, 1);
+  }
+}
+
+function getGrayFromRelLum(relLum: number) {
+  if (relLum <= 0.0031308) {
+    return relLum * 12.92;
+  } else {
+    return (1.0 + 0.055) * Math.pow(relLum, 1.0 / 2.4) - 0.055;
```

Comment:
It looks like you are re-writing the WCAG's luminance and contrast calculator in this file. My thoughts are:

A. This would be good to be pulled into a service instead so it doesn't have to be re-written by other chart types or plugins.
B. EUI does export a `calculateContrast` color service 
C. ChromaJS is a dependency and they have a [`.contrast` method](https://gka.github.io/chroma.js/#chroma-contrast)

Reply:
Looks like a good candidate for the “charts” plugin

Reply:
@cchaos I'm not rewriting the luminance here, I'm inverting the luminance formula to get the lightness to use with the gray color.
This for sure will be part of elastic-charts, I'm currently applying this here because it needs to be released on a patch release and we can't upgrade the chart library in a patch release

Reply:
I can replace `getAAARelativeLum` with the contrast method of `color`
We don't have chromajs as a dependency in Kibana right now

Reply:
updated here: 07f437a

Reply:
++ for moving this to elastic-charts later on